newIDE/app/src/EventsFunctionsExtensionsLoader/CodeWriters/LocalEventsFunctionCodeWriter.js

```diff
@@ -16,7 +16,14 @@ export const makeLocalEventsFunctionCodeWriter = ({
   onWriteFile,
 }: EventsFunctionCodeWriterCallbacks): EventsFunctionCodeWriter => {
   // The generated code for extensions will be stored in a temporary directory
-  const outputDir = os.tmpdir() + '/GDGeneratedEventsFunctions';
+  var outputDirTemp ;
+  try {
+    outputDirTemp = os.tmpdir() + `/GDGeneratedEventsFunctions-${os.userInfo().username}`;
+  }
+  catch (error) {
+    outputDirTemp = os.tmpdir() + `/GDGeneratedEventsFunctions-${os.userInfo().uid}`;
```

Comment:
It's not accessing `username` that will throw, but actually calling `userInfo()`. So this will throw again and crash GDevelop.
Take a step back and consider how you can nicely use a function for this. It should be no more than a few lines.